src/controllers/listings.controller.js

```diff
@@ -6,106 +6,119 @@ import {
-      id: parseInt(id),
-      userId,
-      data: { title, description, price, category }
-    });
-
-    res.status(200).json(updatedListing);
-  } catch (error) {
-    next(error);
+
+  validateRequiredFields({ id }, ['id']);
+
+  const validationError = validateListingData({ id, title, description, price, category, userId });
+  if (validationError) {
+    throw new ValidationError(validationError);
   }
-};
 
-export const getUserListingsAction = async (req, res, next) => {
-  try {
-    const userId = req.user?.id;
-    if (!userId) throw new AppError('User ID is required', 400);
+  const updatedListing = await updateListing({
+    id: parseInt(id),
+    userId,
+    data: { title, description, price, category }
+  });
 
-    const listings = await getUserListings(userId);
-    res.status(200).json(listings);
-  } catch (error) {
-    next(error);
+  if (!updatedListing) {
+    throw new NotFoundError('Listing not found or access denied');
   }
-};
 
-export const deleteListingAction = async (req, res, next) => {
-  try {
-    const userId = req.user?.id;
-    const { id } = req.params;
+  res.status(200).json(updatedListing);
+});
 
-    if (!id || !userId) {
-      throw new AppError('Listing ID and user ID are required', 400);
-    }
+export const getUserListingsAction = asyncHandler(async (req, res) => {
+  const userId = req.user?.id;
 
-    await deleteListing({ id: parseInt(id), userId });
-    res.status(200).json({ message: 'Listing deleted successfully' });
-  } catch (error) {
-    next(error);
+  if (!userId) {
+    throw new ValidationError('User ID is required');
   }
-};
-
-export const toggleListingStatusAction = async (req, res, next) => {
-  try {
-    const userId = req.user?.id;
-    const { id } = req.params;
-
-    if (!id || !userId) {
-      throw new AppError('Listing ID and user ID are required', 400);
-    }
-
-    const updatedListing = await toggleListingStatus({ id: parseInt(id), userId });
-    res.status(200).json({
-      message: `Listing status updated successfully to ${updatedListing.isActive ? 'active' : 'inactive'}`,
-      listing: updatedListing
-    });
-  } catch (error) {
-    next(error);
+
+  const listings = await getUserListings(userId);
+  res.status(200).json(listings);
+});
+
+export const deleteListingAction = asyncHandler(async (req, res) => {
+  const userId = req.user?.id;
+  const { id } = req.params;
+
+  if (!userId) {
+    throw new ValidationError('User ID is required');
   }
-};
 
-export const getInactiveListingsAction = async (req, res, next) => {
-  try {
-    const userId = req.user?.id;
+  validateRequiredFields({ id }, ['id']);
 
-    if (!userId) {
-      throw new AppError('User ID is required', 400);
-    }
+  const deleted = await deleteListing({ id: parseInt(id), userId });
 
-    const inactiveListings = await getInactiveListings(userId);
-    res.status(200).json(inactiveListings);
-  } catch (error) {
-    next(error);
+  if (!deleted) {
```

Comment:
I think you don't need the not found error here. If you delete the already deleted listing, it is fine. just return an ok response.